config/mycnf/mysql57.cnf

```diff
@@ -12,6 +12,9 @@ relay_log_info_repository = TABLE
 relay_log_purge = 1
 relay_log_recovery = 1
 
+# we should never need super privileges
+super-read-only
```

Comment:
While I think this is a good idea — always starting mysqld in super-read-only mode — it does mean that it's something that we'd need to turn off on the primary tablet during the tablet repair. I vaguely recall that we don't do that today?

Reply:
Yeah, we don't touch it in repair: https://github.com/vitessio/vitess/blob/93c811b9f9356e6407c4564cb95bfff9a402f0da/go/vt/vttablet/tabletmanager/rpc_replication.go#L554-L669

Only in init primary: https://github.com/vitessio/vitess/blob/93c811b9f9356e6407c4564cb95bfff9a402f0da/go/vt/vttablet/tabletmanager/rpc_replication.go#L245-L282

And demote primary: https://github.com/vitessio/vitess/blob/93c811b9f9356e6407c4564cb95bfff9a402f0da/go/vt/vttablet/tabletmanager/rpc_replication.go#L378-L469

When this would cause problems is when the mysqld instance for a primary tablet crashes or is otherwise restarted. Ironically, I happen to be doing that in a test I just added here: https://github.com/vitessio/vitess/pull/9308/commits/ef39eb8ed4084c7aa06d3d1740745c9ac4a6b645

Reply:
I'm unsure what the historical reasons were for not fixing read-only in tablet repair, but IMO it would be a good idea.

Reply:
I see what you mean. Tablet restarts and demotes itself to replica through a path that doesn't go through `demotePrimary`. I don't know if there was an actual reason for not doing it. Probably a miss from when `use_super_read_only` was implemented.

Reply:
I looked at the code some more. When the tablet demotes itself, it first calls `demotePrimary` which takes care of read_only (and super_read_only) before `SetReplicationSource` is called. So that path is fine.
The other case where tablet is up but mysqld is down or restarted is something that vtorc will handle. It has never been handled properly prior to vtorc (and replication_reporter was not designed to do that either).

Reply:
As far as the primary goes, we call `SetReadOnly(false)` which also turns off super_read_only (per MySQL docs).